### DISCUSSION TITLE
MacOS and the permissions dilemma

### DISCUSSION THREAD

**Discussion by rrmoore64** [Ylianst/MeshCentral#5064] [General]
I've been running my Mesh server for a few months now and am loving it.

I have several Windows PCs, a handful of Linux, and several more Macs that I have installed.  On the Macs, no matter what I do, I cannot seem to get the Desktop remote control to work most of the time.  It connects, of course, and then shows the background screen and will not let me click or type.  On the user end, it is, of course, asking for permissions to screen recording.  When I go to the system preferences on the target computer and check, meshagent_osx64 is present and checked.  I had manually added it to Screen Recording, Accessibility, and Full Disk Access on every Mac 
 (/usr/local/mesh_services/meshagent/meshagent_osx64)

At this point, I assume there is some OTHER module/file that it wants added to screen recording, maybe?

Thanks,

**Comment by bbrendon** [comment]
From my experience, the screen only works on macOS if it has already been woken up by someone at the console. 

> shows the background screen and will not let me click or type.

This sounds like a permission issue. Check the Accesiblity and screen recording etc. settings.

**Reply by rrmoore64** [reply]
Checked, re-checked, and triple-checked.  Permissions are there for Screen Recording, Full Disk Access, and Accessibility.

I guess the next one that I install I could try for meshagent instead of meshagent_osx64 to see if there is a difference. It is not EVERY Mac - and not EVERY time - but more often than not.  Can't seem to determine a pattern of failures.  All are on Monterey or Ventura - so pretty recent.

**Reply by silversword411** [reply]
Ventura will be extra issues because Apple broke stuff.
https://github.com/Ylianst/MeshCentral/issues/4849

**Comment by bbrendon** [comment]
Also, I never install with  "meshagent_osx64", I use meshagent. I've had problem with that in the past. I'm not sure if they were fixed or not.

**Comment by rrmoore64** [comment]
The only option for download is the 64-bit version now - if you go with the Apple MacOS option.  I guess I could experiment with the Linux / BSD / macOS Binary Installer option and see if it acts differently.

**Comment by xDeepRedx** [comment]
Have the same problem with MacOS Ventura. The agent won't even run on startup on my MacBook with M1. I think MacOS new permissions system doesn't like that it's just an executable and not a .app file and has problems with assigning the permission. Had such a problem with a different application. Please let us know if you find any work-around or solution

**Comment by GPAEALandon** [comment]
We just started using MeshCentral last week and have had some of the same issues. Using the .MPKG, I found when it first installed it would work (after giving permissions), but after a reboot the agent wouldn't re-launch. I was able to manually launch it from Terminal.app, but then I had to grant Terminal permissions in order for it to work. Googling for a better fix, I found an issue reported last November that describes the same issue with a fix. https://github.com/Ylianst/MeshAgent/issues/161#issuecomment-1506600558

**Reply by GPAEALandon** [reply]
Now if they could get the client signed with a developer certificate, we'd be able to push a Configuration Profile that already has permissions granted.
